client/packages/system/src/Patient/ProgramEnrolment/Components/ListView.tsx

```diff
@@ -23,20 +21,11 @@ const ProgramListComponent: FC = () => {
     queryParams: { sortBy, page, first, offset },
   } = useUrlQueryParams();
   const { data, isError, isLoading } = useProgramEnrolment.document.list();
-  const dataWithId: ProgramFragmentWithId[] | undefined = useMemo(
-    () =>
-      data?.nodes.map(node => ({
-        id: node.name,
-        ...node,
-      })),
-    [data]
-  );
   const pagination = { page, first, offset };
   const { localisedDate } = useFormatDateTime();
-  const { setCurrent, setDocumentName, setDocumentType, setProgramType } =
-    usePatientModalStore();
+  const { setCurrent, setDocument, setProgramType } = usePatientModalStore();
 
-  const columns = useColumns<ProgramFragmentWithId>(
+  const columns = useColumns<ProgramRowFragmentWithId>(
```

Comment:
some more not-strictly-necessary refactoring. I was using this object and found it helpful to have the shape already include the `id` prop. Am not using it now, but still think that it is helpful to push the id assignment up